saleor/graphql/product/types/products.py

```diff
@@ -963,7 +963,10 @@ class Product(ChannelContextType[models.Product]):
     product_variants = FilterConnectionField(
         ProductVariantCountableConnection,
         filter=ProductVariantFilterInput(
```

Comment:
I deprecated filter without adding `search` here as it does not make sens - search on product variants looking for products that fit to the provided value either on product or product variants so all product variants are always returned.